src/app/components/search-input-button/search-input-button.controller.js

```diff
@@ -1,14 +1,14 @@
-export default class StarPageController {
+export default class SearchInputButtonController {
     constructor(lastSearchesService, $state) {
         this.lastSearchesService = lastSearchesService;
         this.$state = $state;
     }
 
-    searchCity(city) {
-        if(this.lastSearchesService.saveCity(city)) {
-            this.$state.go('mainPage');
+    searchCity(cityAux) {
+        if (this.lastSearchesService.saveCity(cityAux, this.lastSearchesService.getCities())) {
```

Comment:
it's a good practice retrieve data from a service, save into a variable and then pass it as an argument, instead of call directly it from the arguments. Additionally, the previous name for cityAux was ok, I think that fits better its purpose (because you not have another city variable to need an auxiliary variable), what do you think??